torch/tensor.py

```diff
@@ -441,4 +441,52 @@ def __array_wrap__(self, array):
             array = array.astype('uint8')
         return torch.from_numpy(array)
 
+    @property
+    def __cuda_array_interface__(self):
+        """Array view description for cuda tensors.
+
+        See:
+        https://numba.pydata.org/numba-doc/latest/cuda/cuda_array_interface.html
+        """
+
+        # raise AttributeError for unsupported tensors, so that
+        # hasattr(cpu_tensor, "__cuda_array_interface__") is False.
+        if not self.device.type == "cuda":
+            raise AttributeError(
+                "Not on cuda device, use Tensor.cuda() first: %r" %
```

Comment:
The suggestion to use cuda() seems very odd from my perspective, because I shouldn't be calling __cuda_array_interface__ unless I *knew* that the tensor was a CUDA tensor. If I am calling it on a non-CUDA tensor... well, there is certainly an error, but the fact that you don't have a CUDA tensor doesn't seem like the most salient one.

Reply:
🤔 I chose this phrasing to roughly match the wording of `Tensor.__array__()`, which raises with the message:

`RuntimeError: can't convert CUDA tensor to numpy. Use tensor.cpu() to copy tensor to host memory first.`

I've found that a cpu-tensor-provided-to-cuda-operation (or the inverse) is one of the more common bug types when working with torch, as folks (a) may be coming from libraries which implicitly copy cross device and/or (b) may not fully implement the "allocate on device of your input tensor" design pattern. 

The current behavior of numba is to perform implicit cross-device copies when provided a numpy array and the difference in behavior when moving from `ndarray` to `tensor` may be the first pain point a end-user experiences. For these folks having a explicit callout to `tensor.cuda()` to perform the cross-device copy seems useful. 

More sophisticated users will likely have a deeper understanding of the memory-management semantics of the library, and understand that wantonly running cross-devices transfers isn't a great idea. I think exact exception message will matter less; they'll recognize the cross-device mismatch and begin determining how the issue arose upstream of the call point.

Would you be open to just clarifying the wording to more explicitly matching the `__array__` message?

`AttributeError: can't convert {device.type} tensor to cuda array view. Use tensor.cuda() to copy tensor to device memory first.`

I'm open to alternatives, but I'm not sure what a better phrasing for the exception message would be.

Reply:
That message sounds better.